### PR TITLE
Suggest to install plugins

### PR DESCRIPTION
**Pull request by artem-fedorov**

### CODE REVIEW

File: src/main/java/org/jmeterplugins/repository/Plugin.java [hunk 1]
```diff
@@ -400,4 +420,8 @@ private int compare2(String a, String b) {
     public boolean isVirtual() {
         return markerClass == null;
     }
+
+    public String getMarkerClass() {
```
Comment (undera) [inline src/main/java/org/jmeterplugins/repository/Plugin.java]: Why do we need this method?

File: src/main/java/org/jmeterplugins/repository/Plugin.java [hunk 2]
```diff
@@ -69,6 +70,16 @@ public static Plugin fromJSON(JSONObject elm) {
         if (elm.containsKey("installerClass")) {
             inst.installerClass = elm.getString("installerClass");
         }
+        if (elm.containsKey("componentClasses")) {
```
Comment (undera) [inline src/main/java/org/jmeterplugins/repository/Plugin.java]: Add marker class to it automatically

File: src/main/java/org/jmeterplugins/repository/PluginManager.java [hunk 3]
```diff
@@ -1,25 +1,34 @@
...
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
 
 public class PluginManager {
     private static final Logger log = LoggingManager.getLoggerForClass();
+    private static LoggingHooker hooker = new LoggingHooker();
```
Comment (undera) [inline src/main/java/org/jmeterplugins/repository/PluginManager.java]: is it used anywhere?

File: src/main/java/org/jmeterplugins/repository/Plugin.java (hunk 1 above)
Comment (artem-fedorov) [inline src/main/java/org/jmeterplugins/repository/Plugin.java]: Now we do not need it, we can remove it

File: src/main/java/org/jmeterplugins/repository/PluginManager.java [hunk 4]
```diff
@@ -280,6 +289,7 @@ public void setDoRestart(boolean doRestart) {
         this.doRestart = doRestart;
     }
 
+
```
Comment (undera) [inline src/main/java/org/jmeterplugins/repository/PluginManager.java]: Unneeded change?

File: src/main/java/org/jmeterplugins/repository/PluginManagerDialog.java [hunk 5]
```diff
@@ -11,6 +11,7 @@
 import java.awt.event.ComponentListener;
 import java.io.ByteArrayOutputStream;
 import java.io.PrintStream;
+import java.util.Set;
```
Comment (undera) [inline src/main/java/org/jmeterplugins/repository/PluginManagerDialog.java]: Unneeded change?

File: src/main/java/org/jmeterplugins/repository/plugins/PluginSuggester.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    public TestPlanAnalyzer getAnalyzer() {
+        return analyzer;
+    }
+
+    public void setAnalyzer(TestPlanAnalyzer analyzer) {
+        this.analyzer = analyzer;
+    }
+
+    @Override
+    public void notify(String s) {
```
Comment (undera) [inline src/main/java/org/jmeterplugins/repository/plugins/PluginSuggester.java]: Why do we need this method?

Comment (artem-fedorov) [inline src/main/java/org/jmeterplugins/repository/plugins/PluginSuggester.java]: Because we need GenericCallback for apply changes in pmgr

Comment (undera) [inline src/main/java/org/jmeterplugins/repository/plugins/PluginSuggester.java]: Then, what is magic meaning of `%` sign?

Comment (artem-fedorov) [inline src/main/java/org/jmeterplugins/repository/plugins/PluginSuggester.java]: I see the same notify in PluginManagerCMD

File: src/main/java/org/jmeterplugins/repository/Plugin.java (hunk 2 above)
Comment (artem-fedorov) [inline src/main/java/org/jmeterplugins/repository/Plugin.java]: Added

File: src/main/java/org/jmeterplugins/repository/Plugin.java [hunk 7]
```diff
@@ -69,6 +70,17 @@ public static Plugin fromJSON(JSONObject elm) {
...
+            JSONArray components = elm.getJSONArray("componentClasses");
+            final ArrayList<String> data = new ArrayList<>();
+            data.add(inst.markerClass);
```
Comment (undera) [inline src/main/java/org/jmeterplugins/repository/Plugin.java]: This has to be outside "if contains"

Comment (undera) [inline src/main/java/org/jmeterplugins/repository/Plugin.java]: It needs to be under "if markerClass"

File: src/main/java/org/jmeterplugins/repository/plugins/PluginSuggester.java (hunk 6 above)
Comment (undera) [inline src/main/java/org/jmeterplugins/repository/plugins/PluginSuggester.java]: That's not good answer :) 
This notifier is used to notify user of plugin install progress. Good thing to show in dialog when user agrees to apply changes.

File: src/main/java/org/jmeterplugins/repository/Plugin.java (hunk 7 above)
Comment (artem-fedorov) [inline src/main/java/org/jmeterplugins/repository/Plugin.java]: Fixed

File: src/main/java/org/jmeterplugins/repository/plugins/PluginSuggester.java (hunk 6 above)
Comment (artem-fedorov) [inline src/main/java/org/jmeterplugins/repository/plugins/PluginSuggester.java]: Like the status label at the bottom of pmgr main window?

Comment (undera) [inline src/main/java/org/jmeterplugins/repository/plugins/PluginSuggester.java]: yes, something like that

**Pull request merged**